richtextfx/src/main/java/org/fxmisc/richtext/StyledTextAreaBehavior.java

```diff
@@ -169,13 +169,14 @@
                 .orElse(otherNavigation).ifConsumed((b, e) -> b.view.clearTargetCaretOffset())
                 .orElse(verticalNavigation)
                 .orElse(copyAction)
-                .orElse(charPressConsumer);
+                .orElse(charPressConsumer)
+                .ifConsumed((b, e) -> b.view.requestFollowCaret());
```

Comment:
Maybe these last two lines should be swapped? Shouldn't make a change in observable behavior, but when `charPressConsumer` consumes an event, we expect a KEY_TYPED event, and we `requestFollowCaret` there.